x-pack/plugins/index_management/public/application/components/mappings_editor/components/configuration_form/configuration_form.tsx

```diff
@@ -37,12 +48,30 @@ const formSerializer = (formData: GenericObject, sourceFieldMode?: string) => {
     ? 'strict'
     : dynamicMapping?.enabled;
 
+  const _source =
```

Comment:
nit: It might just be me, but this seems a little too complex to read at first glance. Perhaps a more explicit structure could make future extensions more straightforward? 🤔